### ISSUE TITLE
Cannot flash to the board

### ISSUE THREAD

**Issue by sheldon123z** [aws/amazon-freertos#870]
Hi,
I am super confused about the documents. It has been changed many times in the past week.  I am using esp32 DevKit to do a small project with freeRTOS, and the last release was very smooth for me. However, this release migrated to Cmake build system, and then many problems have occurred. I got many suggestions from @tgsong in another issue, but my problem hasn't been solved. I noticed the newest docs deleted the part about using ninja to build the application, and actually, the ninja build was never effected for me. I used `cmake --build  build --target flash` to flash the demo to the board, but after everything was done, nothing happens to my IoT console. The program stopped at the last step. 
Here is the complete building-flashing process:

------------------------------------------record---------------------------------------------------------
Microsoft Windows [Version 10.0.17134.829]
(c) 2018 Microsoft Corporation. All rights reserved.

c:\Users\shzheng\Desktop>cd FreeRTOSWindows\amazon-freertos-master

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master>cmake --build build --target flash
[0/2] Re-checking globbed directories...
[1/4] Performing build step for 'bootloader'
ninja: no work to do.
[1/2] cmd.exe /C "cd /D C:\Users\shzheng\Desktop\FreeRTOSW...indows/amazon-freertos-master/build" -P run_esptool.cmake"
Note: esptool.py will search for a serial port. To specify a port, set the ESPPORT environment variable.
Note: Using default baud rate 460800. To modify, set ESPBAUD environment variable.

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master>cmake --build build4 --target flash
[0/2] Re-checking globbed directories...
[1/4] Performing build step for 'bootloader'
ninja: no work to do.
[1/2] cmd.exe /C "cd /D C:\Users\shzheng\Desktop\FreeRTOSW...ndows/amazon-freertos-master/build4" -P run_esptool.cmake"
Note: esptool.py will search for a serial port. To specify a port, set the ESPPORT environment variable.
Note: Using default baud rate 460800. To modify, set ESPBAUD environment variable.

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master>aws --version
aws-cli/1.16.184 Python/3.6.0 Windows/10 botocore/1.12.174

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master>cmake -DVENDOR=espressif -DBOARD=esp32_wrover_kit -DCOMPILER=xtensa-esp32 -GNinja -S . -B build4
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Program Files/Espressif/ESP-IDF Tools/toolchain/bin/xtensa-esp32-elf-gcc.exe
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.21.0.windows.1")
-- Checking Python dependencies...
Python requirements from C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/requirements.txt are satisfied.
-- Building for target esp32
-- Component names: soc log heap freertos vfs newlib driver tcpip_adapter xtensa-debug-module app_trace partition_table app_update spi_flash mbedtls micro-ecc bootloader_support ethernet nvs_flash pthread smartconfig_ack wpa_supplicant esp32 cxx amazon-freertos-common bootloader bt console esp_adc_cal esptool_py expat wear_levelling sdmmc fatfs nghttp openssl spiffs ulp
-- Component paths: C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/soc;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/log;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/heap;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/freertos;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/vfs;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/newlib;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/driver;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/tcpip_adapter;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/xtensa-debug-module;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/app_trace;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/partition_table;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/app_update;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/spi_flash;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/mbedtls;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/micro-ecc;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader_support;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/ethernet;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/nvs_flash;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/pthread;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/smartconfig_ack;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/wpa_supplicant;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/cxx;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/amazon-freertos-common;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bt;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/console;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp_adc_cal;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esptool_py;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/expat;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/wear_levelling;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/sdmmc;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/fatfs;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/nghttp;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/openssl;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/spiffs;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/ulp
Creating config file C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/build4/sdkconfig...
-- IDF_VER: v3.1.3-25-g36733db8d
fatal: not a git repository (or any of the parent directories): .git
-- Project version: HEAD-HASH-NOTFOUND
fatal: not a git repository (or any of the parent directories): .git
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/build4/vendors/espressif/boards/esp32/esp-idf/esp32/esp32_out.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.common.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.rom.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.peripherals.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.rom.spiram_incompatible_fns.ld
-- Component libraries:
=========================Resolving dependencies==========================
module disabled: posix
reason: posix::mcu_port is not defined by vendor.
dependency path: posix->posix::mcu_port

====================Configuration for Amazon FreeRTOS====================
Version: 201906.00 Major
Git version: Unknown

Target microcontroller:
vendor: Espressif
board: ESP-WROVER-KIT
description: Development board produced by Espressif that has support for
LCD, MicroSD card, and USB JTAG interface. It comes in two
variants either with ESP-WROOM-32 or ESP32-WROVER module.
family: ESP32
data ram size: 520KB
program memory size: 4MB

Host platform:
OS: Windows-10.0.17134
Toolchain: xtensa-esp32
Toolchain path: C:/Program Files/Espressif/ESP-IDF Tools/toolchain
CMake generator: Ninja

Amazon FreeRTOS modules:
Modules to build: ble, ble_wifi_provisioning, common, crypto, defender,
dev_mode_key_provisioning, freertos_plus_tcp, greengrass,
kernel, mqtt, ota, pkcs11, platform, secure_sockets,
serializer, shadow, tls, wifi
Enabled by user: ble, ble_wifi_provisioning, defender, greengrass, mqtt, ota,
pkcs11, platform, secure_sockets, shadow, wifi
Enabled by dependency: ble_hal, common, crypto, demo_base, dev_mode_key_provisioning,
freertos, freertos_plus_tcp, kernel, pkcs11_mbedtls,
secure_sockets_freertos_plus_tcp, serializer, tls, utils
3rdparty dependencies: jsmn, mbedtls, pkcs11, tinycbor
Available demos: demo_ble, demo_ble_numeric_comparison, demo_defender,
demo_greengrass_connectivity, demo_mqtt, demo_ota,
demo_shadow, demo_tcp, demo_wifi_provisioning
Available tests:
==================================.=======================================

-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/build4

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master>ninja
ninja: error: loading 'build.ninja': The system cannot find the file specified.

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master>cmake --build build4
[0/2] Re-checking globbed directories...
[135/861] Building C object vendors/espressif/boards/esp32...s.dir/////demos/ble/aws_ble_gatt_server_demo.c.obj
../demos/ble/aws_ble_gatt_server_demo.c:273:19: warning: 'vGattDemoSvcHook' defined but not used [-Wunused-function]
static BaseType_t vGattDemoSvcHook( void )
^
[170/861] Building C object vendors/espressif/boards/esp32...sp32/CMakeFiles/idf_component_esp32.dir/hwcrypto/sha.c.obj
../vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c: In function 'esp_sha_unlock_engine':
../vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c:201:39: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
SemaphoreHandle_t *engine_state = sha_get_engine_state(sha_type);
^
../vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c: In function 'esp_sha_read_digest_state':
../vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c:234:43: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
SemaphoreHandle_t *engine_state = sha_get_engine_state(sha_type);
^
../vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c: In function 'esp_sha_block':
../vendors/espressif/esp-idf/components/esp32/hwcrypto/sha.c:269:43: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
SemaphoreHandle_t engine_state = sha_get_engine_state(sha_type);
^
[171/861] Building C object vendors/espressif/boards/esp32...2/CMakeFiles/idf_component_esp32.dir/wifi_os_adapter.c.obj
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:423:22: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._queue_delete = vQueueDelete,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:423:22: note: (near initialization for 'g_wifi_osi_funcs._queue_delete')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:429:29: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._queue_recv_from_isr = xQueueReceiveFromISR,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:429:29: note: (near initialization for 'g_wifi_osi_funcs._queue_recv_from_isr')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:430:27: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._queue_msg_waiting = uxQueueMessagesWaiting,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:430:27: note: (near initialization for 'g_wifi_osi_funcs._queue_msg_waiting')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:431:28: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._event_group_create = xEventGroupCreate,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:431:28: note: (near initialization for 'g_wifi_osi_funcs._event_group_create')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:432:28: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._event_group_delete = vEventGroupDelete,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:432:28: note: (near initialization for 'g_wifi_osi_funcs._event_group_delete')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:433:30: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._event_group_set_bits = xEventGroupSetBits,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:433:30: note: (near initialization for 'g_wifi_osi_funcs._event_group_set_bits')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:434:32: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._event_group_clear_bits = xEventGroupClearBits,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:434:32: note: (near initialization for 'g_wifi_osi_funcs._event_group_clear_bits')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:438:21: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._task_delete = vTaskDelete,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:438:21: note: (near initialization for 'g_wifi_osi_funcs._task_delete')
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:441:31: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
._task_get_current_task = xTaskGetCurrentTaskHandle,
^
../vendors/espressif/esp-idf/components/esp32/wifi_os_adapter.c:441:31: note: (near initialization for 'g_wifi_osi_funcs._task_get_current_task')
[299/861] Building C object vendors/espressif/boards/esp32...river/CMakeFiles/idf_component_driver.dir/rtc_module.c.obj
../vendors/espressif/esp-idf/components/driver/rtc_module.c: In function 'touch_pad_filter_start':
../vendors/espressif/esp-idf/components/driver/rtc_module.c:964:15: warning: passing argument 5 of 'xTimerCreate' from incompatible pointer type [-Wincompatible-pointer-types]
NULL, touch_pad_filter_cb);
^
In file included from ../vendors/espressif/boards/esp32/aws_demos/application_code/espressif_code/freertos/include/freertos/timers.h:1:0,
from ../vendors/espressif/esp-idf/components/driver/rtc_module.c:36:
../freertos_kernel/include/timers.h:228:16: note: expected 'TimerCallbackFunction_t {aka void ()(struct tmrTimerControl )}' but argument is of type 'void ()(void *)'
TimerHandle_t xTimerCreate( const char * const pcTimerName, /*lint !e971 Unqualified char types are allowed for strings and single characters only. */
^
[474/861] Performing configure step for 'bootloader'
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.21.0.windows.1")
-- IDF_TARGET not set, using default target: esp32
-- Building for target esp32
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Program Files/Espressif/ESP-IDF Tools/toolchain/bin/xtensa-esp32-elf-gcc.exe
-- Component names: soc esp32 log bootloader spi_flash micro-ecc bootloader_support main esptool_py partition_table
-- Component paths: C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/soc;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/log;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/spi_flash;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/micro-ecc;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader_support;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader/subproject/main;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esptool_py;C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/partition_table
-- Test component names:
-- Test component paths:
-- IDF_VER: v3.1.3-25-g36733db8d
fatal: not a git repository (or any of the parent directories): .git
-- Project version: HEAD-HASH-NOTFOUND
fatal: not a git repository (or any of the parent directories): .git
-- Component libraries:
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader/subproject/main/esp32.bootloader.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/bootloader/subproject/main/esp32.bootloader.rom.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.rom.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.rom.spiram_incompatible_fns.ld
-- Adding linker script C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/vendors/espressif/esp-idf/components/esp32/ld/esp32.peripherals.ld
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/shzheng/Desktop/FreeRTOSWindows/amazon-freertos-master/build4/bootloader
[528/861] Performing build step for 'bootloader'
[1/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/soc_memory_layout.c.obj
[2/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/gpio_periph.c.obj
[3/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/rtc_periph.c.obj
[4/38] Generating dummy_main_src.c
[5/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/rtc_pm.c.obj
[6/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/rtc_init.c.obj
[7/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/cpu_util.c.obj
[8/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/rtc_sleep.c.obj
[9/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/rtc_time.c.obj
[10/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/rtc_clk.c.obj
[11/38] Building C object CMakeFiles/bootloader.elf.dir/dummy_main_src.c.obj
[12/38] Building C object esp-idf/soc/CMakeFiles/idf_component_soc.dir/esp32/spi_periph.c.obj
[13/38] Building C object esp-idf/spi_flash/CMakeFiles/idf_component_spi_flash.dir/spi_flash_rom_patch.c.obj
[14/38] Building C object esp-idf/log/CMakeFiles/idf_component_log.dir/log.c.obj
[15/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_random.c.obj
[16/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_common.c.obj
[17/38] Building C object esp-idf/micro-ecc/CMakeFiles/idf_component_micro-ecc.dir/micro-ecc/uECC.c.obj
[18/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_flash.c.obj
[19/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_sha.c.obj
[20/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/efuse.c.obj
[21/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_clock.c.obj
[22/38] Linking C static library esp-idf\log\liblog.a
[23/38] Linking C static library esp-idf\spi_flash\libspi_flash.a
[24/38] Linking C static library esp-idf\soc\libsoc.a
[25/38] Linking C static library esp-idf\micro-ecc\libmicro-ecc.a
[26/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_utility.c.obj
[27/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/flash_partitions.c.obj
[28/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/flash_encrypt.c.obj
[29/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/flash_qio_mode.c.obj
[30/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/secure_boot_signatures.c.obj
[31/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/secure_boot.c.obj
[32/38] Building C object esp-idf/main/CMakeFiles/idf_component_main.dir/bootloader_start.c.obj
[33/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/esp_image_format.c.obj
[34/38] Building C object esp-idf/bootloader_support/CMakeFiles/idf_component_bootloader_support.dir/src/bootloader_init.c.obj
[35/38] Linking C static library esp-idf\main\libmain.a
[36/38] Linking C static library esp-idf\bootloader_support\libbootloader_support.a
[37/38] Linking C executable bootloader.elf
[38/38] Generating ../bootloader.bin
esptool.py v2.6
[549/861] Building C object vendors/espressif/boards/esp32...Files/idf_component_bt.dir/bluedroid/osi/fixed_queue.c.obj
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c: In function 'fixed_queue_try_dequeue':
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c:192:22: warning: passing argument 1 of 'osi_sem_take' from incompatible pointer type [-Wincompatible-pointer-types]
if (osi_sem_take(queue->dequeue_sem, 0) != 0) {
^
In file included from ../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c:26:0:
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/include/osi/semaphore.h:38:5: note: expected 'struct QueueDefinition **' but argument is of type 'osi_sem_t {aka struct QueueDefinition *}'
int osi_sem_take(osi_sem_t *sem, uint32_t timeout);
^
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c: In function 'fixed_queue_try_remove_from_queue':
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c:246:26: warning: passing argument 1 of 'osi_sem_take' from incompatible pointer type [-Wincompatible-pointer-types]
osi_sem_take(queue->dequeue_sem, 0) == 0) {
^
In file included from ../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c:26:0:
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/include/osi/semaphore.h:38:5: note: expected 'struct QueueDefinition **' but argument is of type 'osi_sem_t {aka struct QueueDefinition *}'
int osi_sem_take(osi_sem_t *sem, uint32_t timeout);
^
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c:253:22: warning: passing argument 1 of 'osi_sem_give' from incompatible pointer type [-Wincompatible-pointer-types]
osi_sem_give(queue->enqueue_sem);
^
In file included from ../vendors/espressif/esp-idf/components/bt/bluedroid/osi/fixed_queue.c:26:0:
../vendors/espressif/esp-idf/components/bt/bluedroid/osi/include/osi/semaphore.h:40:6: note: expected 'struct QueueDefinition **' but argument is of type 'osi_sem_t {aka struct QueueDefinition *}'
void osi_sem_give(osi_sem_t *sem);
^
[861/861] Generating ../../../../../aws_demos.bin
esptool.py v2.6

[0/2] Re-checking globbed directories...
[1/4] Performing build step for 'bootloader'
ninja: no work to do.
[1/2] cmd.exe /C "cd /D C:\Users\shzheng\Desktop\FreeRTOSW...indows/amazon-freertos-master/build" -P run_esptool.cmake"
Note: esptool.py will search for a serial port. To specify a port, set the ESPPORT environment variable.
Note: Using default baud rate 460800. To modify, set ESPBAUD environment variable.

c:\Users\shzheng\Desktop\FreeRTOSWindows\amazon-freertos-master> 
------------------------------------end----------------------------------------------------------------

As you can see, it should show some steps of connecting to the IoT console, but it didn't happen.
I also tried it on MacOS, but for some unknown reasons, the board won't connect to my computer(the serial connection failed, but I installed the driver). 
Please help me and point out which step is wrong, I really need help with it.

Many thanks in advance!
Sheldon

**Comment by alexa-noxon** [comment]
Hi Sheldon,

Apologies for the documentation changes-- we are working to update the docs and it looks like we have a little more work yet to do still.

Can you tell if the aws_demos.bin file is actually being generated?  This step might just have only generated the elf file (even though the print statement indicates otherwise).   If that is the case and there is no .bin, you will need to run the esptool to covert from elf to bin.

`python \path-to-repo\amazon-freertos\vendors\espressif\esp-idf\components\esptool_py\esptool\esptool.py --chip esp32 elf2image aws_demos
esptool.py v2.6`

and do the same for the bootloader
`cd bootloader`
`python \path-to-repo\amazon-freertos\vendors\espressif\esp-idf\components\esptool_py\esptool\esptool.py --chip esp32 elf2image bootloader.elf
esptool.py v2.6`

Then you would be able to pick back up with the instructions for using the IDF python tool idf.py to erase the flash, flash with a new image, and get the monitor output
`./vendors/espressif/esp-idf/tools/idf.py erase_flash flash monitor -p COMxy -B build`

Let me know if that resolves the issue, and if not, whether you can actually see the .bin file in the file system or not.   In the meantime we will work to update the documentation (again!) to capture all of these steps.

Thanks for reporting this,
Alexa

**Comment by sheldon123z** [comment]
Hi Alexa,
Thank you for your reply. I have checked the files, and there are the .bin files in the folder. I just want to quit windows now and try MacOS. But I don't know why my Macbook pro cannot find the esp32 board. I have installed the driver, and the device info showed the driver was there. My mac has four thunderbolt type C port, I am wondering if that is the reason.

By the way, I can't run the idf.py on my windows, the system prompts the directory is invalid.

**Comment by sarenameas** [comment]
Great that you can flash the board now. 

Try erasing the flash then starting over:
`python ./vendors/espressif/esp-idf/tools/idf.py erase_flash -B build`

Please make sure that your thing's certificate and private key are pasted into: 
https://github.com/aws/amazon-freertos/blob/master/demos/include/aws_clientcredential_keys.h

Please use try the ATS endpoint: `a1zwfs4sf2i9un-ats.iot.us-east-1.amazonaws.com`

Also will need an asterisk * at the end of your resource strings:
```
{
"Version": "2012-10-17",
"Statement": [
{
"Effect": "Allow",
"Action": "iot:Connect",
"Resource": "arn:aws:iot:us-east-1:330321474979:*"
},
{
"Effect": "Allow",
"Action": "iot:Publish",
"Resource": "arn:aws:iot:us-east-1:330321474979:*"
},
{
"Effect": "Allow",
"Action": "iot:Subscribe",
"Resource": "arn:aws:iot:us-east-1:330321474979:*"
},
{
"Effect": "Allow",
"Action": "iot:Receive",
"Resource": "arn:aws:iot:us-east-1:330321474979:*"
}
]
}
```

**Comment by sheldon123z** [comment]
Hi Sarenameas，

Thank you very much for your reply. I have found the problem that my board cannot communicate to aws iot console. The reason is that my company blocked the AWS iot port. After I used my phone's wifi hotspot, everything went well.  

Issue summary: I have tried both Windows and MacOS operating system, and found out that the freeRTOS is so hard to setup on Windows, I struggled for a week but haven't succeeded yet. The ninja build system is pretty hard to use because the lack of instructions. Also, under MacOS, the esp32 driver needs to be removed from the blocking list of the security system, or the computer won't detect the board.

Thank you very much and this issue is closed.
